bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/ExtensibleTrustManagerImpl.java

```diff
@@ -158,24 +173,24 @@ private String getCommonName(X509Certificate x509Certificate) {
     }
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    protected void addTlsCertificateProvider(TlsCertificateProvider tlsCertificateProvider) {
+    public void addTlsCertificateProvider(TlsCertificateProvider tlsCertificateProvider) {
         X509ExtendedTrustManager trustManager = new TlsCertificateTrustManagerAdapter(tlsCertificateProvider)
                 .getTrustManager();
         mappingFromTlsCertificateProvider.put(tlsCertificateProvider, trustManager);
         addLinkedTrustManager(tlsCertificateProvider.getHostName(), trustManager);
     }
 
-    protected void removeTlsCertificateProvider(TlsCertificateProvider tlsCertificateProvider) {
+    public void removeTlsCertificateProvider(TlsCertificateProvider tlsCertificateProvider) {
         removeLinkedTrustManager(tlsCertificateProvider.getHostName(),
                 mappingFromTlsCertificateProvider.remove(tlsCertificateProvider));
     }
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    protected void addTlsTrustManagerProvider(TlsTrustManagerProvider tlsTrustManagerProvider) {
+    public void addTlsTrustManagerProvider(TlsTrustManagerProvider tlsTrustManagerProvider) {
         addLinkedTrustManager(tlsTrustManagerProvider.getHostName(), tlsTrustManagerProvider.getTrustManager());
```

Comment:
Ideally it should not be, maybe we should state in the interface that object should be immutable.

Given this class is checked for every http call the efficiency is really important, so I want to be able to look something up in in `O(n)`

Reply:
I agree. I need to look deeper into the order of operations when a thing config changes. This will determine which piece should be responsible for adding / removing trust managers.

Reply:
I added a remark in the interface that implementation should be immutable.